diff -ruN tailscale_old/cmd/derper/cert.go tailscale/cmd/derper/cert.go
--- tailscale_old/cmd/derper/cert.go	2024-12-08 01:03:58.444963207 +0800
+++ tailscale/cmd/derper/cert.go	2024-12-08 01:04:13.188872566 +0800
@@ -88,9 +88,9 @@
 }
 
 func (m *manualCertManager) getCertificate(hi *tls.ClientHelloInfo) (*tls.Certificate, error) {
-	if hi.ServerName != m.hostname {
-		return nil, fmt.Errorf("cert mismatch with hostname: %q", hi.ServerName)
-	}
+//	if hi.ServerName != m.hostname {
+//		return nil, fmt.Errorf("cert mismatch with hostname: %q", hi.ServerName)
+//	}
 
 	// Return a shallow copy of the cert so the caller can append to its
 	// Certificate field.
